### DISCUSSION TITLE
How do I add a referer to an HTTP request? I added it directly in the request header and it doesn't seem to work.

### DISCUSSION THREAD

**Discussion by 2394024600** [AutomaApp/automa#1815] [Q&A]
Such as the title

**Comment by 2394024600** [comment]
![1725373286597](https://github.com/user-attachments/assets/697fd7db-4e15-451e-98a1-099b6f9165d8)
In postman, I can add referer directly to the request header, but not in automa. Who can help me? Thank you very much.

**Comment by jingbof** [comment]
**TL;DR:** 

1. If you're executing JS block as a background script (execution context = Background), then yes, it's possible to set a custom header.

2. If you're executing JS block on an active tab (execution context = Active Tab), then it's not possible.

This is because when a script is executed from a webpage, the requests initiated directly by the webpage itself (e.g., through JavaScript like `fetch`) will typically carry the headers defined by that webpage or the browser itself. As a content script in a Chrome extension, you cannot directly modify the headers of those network requests initiated by the webpage due to the browser's security and privacy policies.

To set a custom header with a background script, you can use something like:

```javascript
fetch('https://api.example.com/endpoint', {
  method: 'POST',
  headers: {
    'Authorization': 'Bearer your-token',
    'Custom-Header': 'CustomHeaderValue'
  },
  body: JSON.stringify({
    key: 'value'
  })
})
  .then(response => response.json())
  .then(data => console.log(data))
  .catch(error => console.error('Error:', error));
```
